x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/PasswordAndEnrollmentInitialNode.java

```diff
@@ -0,0 +1,239 @@
+import org.elasticsearch.common.settings.SecureSetting;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.xpack.core.XPackSettings;
+import org.elasticsearch.xpack.core.security.user.ElasticUser;
+import org.elasticsearch.xpack.security.tool.CommandLineHttpClient;
+import org.elasticsearch.xpack.security.tool.HttpResponse;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.xpack.security.authc.esnative.tool.SetupPasswordTool.getErrorCause;
+
+public class PasswordAndEnrollmentInitialNode extends EnvironmentAwareCommand {
+    private static final Setting<SecureString> bootstrapPassword = SecureSetting.secureString("bootstrap.password",
+        null);
+    private static final Setting<SecureString> credentialsPassword = SecureSetting.secureString("bootstrap.password",
+        KeyStoreWrapper.SEED_SETTING);
+    private static final String elasticUser = ElasticUser.NAME;
+
+    private SecureString password;
+    private String token;
+    private String fingerprint;
+
+    // Protected for testing
+    protected SecureString getPassword() {
+        return password;
+    }
+
+    protected String getToken() {
+        return token;
+    }
+
+    protected String getFingerprint() {
+        return fingerprint;
+    }
+
+    protected CommandLineHttpClient getClient(Environment env) {
+        return new CommandLineHttpClient(env);
+    }
+
+    protected CreateEnrollmentToken getCreateEnrollmentToken(Environment env) throws Exception{
+        return new CreateEnrollmentToken(env);
+    }
+
+
+    PasswordAndEnrollmentInitialNode (){
+        super("Set elastic password and generate enrollment token for initial node");
+        parser.allowsUnrecognizedOptions();
+    }
+
+    public static void main(String[] args) throws Exception {
+        exit(new PasswordAndEnrollmentInitialNode().main(args, Terminal.DEFAULT));
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        if (options.nonOptionArguments().contains("--explicitly-acknowledge-execution") == false) {
+            throw new UserException(ExitCodes.NOOP, "This command is not intended for end users");
```

Comment:
My thoughts when introducing this error code were for it to be used when requesting the creation of something that already exists. For the initial Security autoconfiguration the idea was that if Security is already configured then the autoconfiguration is a noop.

I would pick `ExistCodes.CONFIG` for this particular case.